Large reorganisation & refactor

zanderwar opened the pull request.

zanderwar committed:
Update to include cache rule guidlines

zanderwar committed:
Added /dev/ to no-cache recommendations

Commit:
WIP:
Split out the Purge functionality into CloudFlare_Purge where it now takes on an OOP role
Split out the notifications functionality into CloudFlare_Notifications
Split out the errorhandlers into CloudFlare_ErrorHandlers
Split the tests up where relevant

Commit:
Merge remote-tracking branch 'origin/enhancement-dry_purge' into enhancement-dry_purge

# Conflicts:
#	README.md

robbieaverill commented on code/CloudFlare.php:
```diff
@@ -59,241 +63,102 @@ public function getCFCredentials()
-                "Successfully purged <strong>EVERYTHING</strong> from cache."
-            )
-        );
+        return CloudFlareAdmin::singleton()->purge_all();
```
I don't like this. If I didn't want the admin model then this would break my purge all capability

zanderwar replied:
Noted, the intention is to remove these functions entirely, this PR is a WIP (forgot to tag it sorry) I do intend to still add the centralised functionality for purge*. Was just trying to keep some backwards compatibility without repeating myself pointlessly.

robbieaverill commented on code/CloudFlare.php:
```diff
@@ -59,241 +63,102 @@ public function getCFCredentials()
-        );
+        Deprecation::notice('2.0', 'This method will be removed in favor for CloudFlare_Purge functionality');
 
+        return CloudFlareAdmin::singleton()->purge_images();
```
Maybe move the functions into this model instead?

zanderwar replied:
I considered the possibility of branching out from just cache purging into routes etc, ultimately I'd like to keep any purge functionality seperate, I believe that function is also void too so not sure why I bothered returning

robbieaverill commented on code/CloudFlare.php:
```diff
@@ -407,93 +272,33 @@ public function fetchZoneID()
-     * @param null       $isRecursing
+     * @param string $method
      *
+     * @param null $isRecursing
```
This should probably reflect the expected value rather than the fallback

robbieaverill commented on code/CloudFlare.php:
```diff
@@ -828,11 +534,51 @@ public function curlRequest($url, $data = null, $method = 'DELETE')
+            "Content-Type: application/json"
+        );
+
+        $this->extend("updateCloudFlareAuthHeaders", $headers);
```
Nice :-D

robbieaverill commented on code/CloudFlare_ErrorHandlers.php:
```diff
@@ -0,0 +1,35 @@
+
+    public static function malformed($response)
+    {
+        self::log(sprintf("The API response was malformed:\r\n%s", print_r($response)));
```
Needs true as second argument

robbieaverill commented on code/CloudFlare_Notifications.php:
```diff
@@ -0,0 +1,58 @@
+        $jar['CFMessage'] = $message;
+        $jar['CFType'] = $type;
+
+        CloudFlare::singleton()->setSessionJar($jar);
```
Seems a bit odd to me that you kind of reference these classes in a circular manner

robbieaverill replied:
Ignore this, not the Purge class

robbieaverill commented on code/CloudFlare_Purge.php:
```diff
@@ -0,0 +1,433 @@
+        }
+
+        return $this;
+    }
```
You could probably do away with one of these two methods and just use one for everything

robbieaverill commented on code/CloudFlare_Purge.php:
```diff
@@ -0,0 +1,433 @@
+                )
+            );
+        }
+
```
This seems like it should be one method call with one message, they're basically identical

Commit:
Merge branch 'master' into enhancement-dry_purge

# Conflicts:
#	code/CloudFlare.php
#	code/extensions/CloudFlareExt.php

robbieaverill commented on code/CloudFlare_Purge.php:
```diff
@@ -0,0 +1,433 @@
+    }
+
+    /**
+     * @return mixed
```
array only?

zanderwar replied:
Yeah it is array only \*glares at phpstorm phpdoc generator*. Mixed is not very helpful I personally never use it :)

robbieaverill commented on code/CloudFlare_Purge.php:
```diff
@@ -0,0 +1,433 @@
+    }
+
+    /**
+     * @param null $bool
```
Boolean - if you passed null it wouldn't be expected use

robbieaverill commented on code/CloudFlare_Purge.php:
```diff
@@ -0,0 +1,433 @@
+     * @param null $bool
+     * @return $this
+     */
+    public function purgeEverything($bool = null)
```
Can you call this setPurgeEverything?

robbieaverill commented on code/CloudFlare_Purge.php:
```diff
@@ -0,0 +1,433 @@
+    /**
+     * @param $response
+     */
+    private function setResponse($response)
```
Private sucks. Protected allows for overloading

zanderwar replied:
Yeah I didn't really see an API use for it, but protected it shall be if you see otherwise  :)

robbieaverill replied:
The way I see it is it should only be private if you have a reason why it should be private :-) that's just my opinion though

robbieaverill reviewed (commented):
Just some comments. Take what you want to from them :-)

zanderwar committed:
Changes:
- Some changes as per PR review
- New tests
- Removed dependency of CloudFlareAdmin for deprecated functions
- Removed pushFiles() and consolidated into pushFile()

zanderwar commented on code/CloudFlare_Purge.php:
```diff
@@ -121,26 +123,35 @@ private function rglob($pattern, $flags = 0)
+            if (basename($dir) == 'framework') {
+                continue;
+            }
+            // if we implode the array we should be able to search for all extensions at once like below
```
This is definitely a priority, the more extensions to search for the longer the request takes.

Windows 10 WAMP PHP 5.5 - Unable to get any results from the GLOB_BRACE approach

zanderwar replied:
bump

zanderwar committed:
PHP Doc Fixes

zanderwar committed:
Restored function omitted in merge conflicts

zanderwar committed:
i18n and refactor fixes

zanderwar committed:
print_r fix to return

zanderwar commented on code/CloudFlare_Purge.php:
```diff
@@ -0,0 +1,531 @@
+            // if we implode the array we should be able to search for all extensions at once like below
+            // http://stackoverflow.com/a/23969253/2266583 w/ GLOB_BRACE
+            // however it would always return no files for me, so i'll just leave it here for now
+            // $files = array_merge($this->rglob(rtrim($rootDir, "/") . '/*.{'.$extensions.'}'), $files);
```
This is definitely a priority, the more extensions to search for the longer the request takes.

Windows 10 WAMP PHP 5.5 - Unable to get any results from the GLOB_BRACE approach

zanderwar committed:
Renamed class to conform

zanderwar committed:
Fixed a bug where the baseUrl would removed when converting from document root to the absolute url

zanderwar committed:
Success/Failure Messages (Race condition with CMS however)

zanderwar committed:
Fix for single page purge from MoreOptions menu

zanderwar commented on code/extensions/CloudFlareExtension.php:
```diff
@@ -19,7 +19,7 @@ public function onAfterPublish(&$original)
             $purger = CloudFlare_Purge::create();
             $shouldPurgeRelations = CloudFlare_Purge::singleton()->getShouldPurgeRelations();
-            $urls = array(DataObject::get_by_id("SiteTree", $this->owner->ID)->Link());
+            $urls = array($_SERVER['DOCUMENT_ROOT'] . ltrim(DataObject::get_by_id("SiteTree", $this->owner->ID)->Link(), "/"));
```
@robbieaverill Just a headsup

zanderwar commented on code/CloudFlare_Purge.php:
```diff
@@ -479,7 +479,7 @@ public function quick($what, $other_id = null) {
 
             $purger
-                ->pushFile($page->Link())
+                ->pushFile(str_replace("//","/",$_SERVER['DOCUMENT_ROOT'] . "/" .$page->Link()))
```
@robbieaverill  and another

zanderwar commented on code/extensions/CloudFlareExtension.php:
```diff
@@ -89,6 +89,7 @@ public function onAfterPublish(&$original)
 
         }
 
+        //die(var_dump($purger->getFiles()));
```
debug

zanderwar committed:
Removed debug

zanderwar commented:
I'm happy to merge this in now, everything tests OK on my end.

zanderwar commented:
Resolves #12
